Don't display empty tooltips on popup menus.

bambinoua opened the pull request:
Sometimes it is necessary to suppress the tooltip for a popup menu but it was impossible. Even if tooltip property was set to empty string the ugly tooltip was shown. This minor fix allows to use empty string to suppress the tooltip. It makes sense because empty string to show is useless.

bambinoua committed:
Update popup_menu.dart (from line 1167 to 1182)

Sometimes it is necessary to suppress the tooltip for a menu but it was impossible. Even if tooltip property was set to empty string the ugly tooltip was shown. This minor fix allows to use empty string to suppress the tooltip. It makes sense because empty string to show is useless.

bambinoua committed:
Update popup_menu.dart (lines 1168 and 1170)

Modification to respect a rule `always_specify_types`.

bambinoua committed:
Update popup_menu.dart

Minor code refactoring.

bambinoua committed:
Update popup_menu.dart

Specify type for tooltip variable

gspencergoog commented:
This seems like an interesting idea, but I think it could be more general: could you instead make the `Tooltip` widget just not display itself when the tooltip message was empty? That seems generally useful, since I can't think of why anyone would want to actually display an empty tooltip. The `_TooltipState.build` could just return a `SizedBox(child: widget.child)` if the message was empty.

bambinoua committed:
Upadate tooltip.dart (lines 475-479)

The `Tooltip` widget does not allow null value but it allows empty string value. In this case user see the small dark rectangle with no text which is useless. On basis of this suggestion https://github.com/flutter/flutter/pull/86165#issuecomment-881039330 I added additional check if tooltip message is empty then just return the wrapped widget.

bambinoua committed:
Update tooltip.dart

Fix trailing spaces

gspencergoog commented on packages/flutter/lib/src/material/tooltip.dart:
```diff
@@ -472,6 +472,11 @@ class _TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
+    // If message is empty then no need to create a tooltip overlay to show
+    // the empty black container so just return the wrapped child as is
+    if (widget.message.isEmpty) {
+      return widget.child!;
```
I think that it's pretty often that the tooltip wouldn't have a child.  Perhaps you could do something like this instead, so that it wouldn't crash if there were no child.

```
return widget.child ?? const SizedBox();
```

bambinoua replied:
You are absolutely right. I was a little innattentive and didn't noticed that `child` is not required parameter.

bambinoua committed:
Update tooltip.dart

As tooltip's `child` can be `null` empty `SizedBox` will be returned when tooltip's `message` is empty.

bambinoua commented:
@gspencergoog it looks like one check is failed and I think it is because I have outdated `tooltip.dart` file. Could you please to advise me what I need to to to fix this `ci.yaml validation`?

bambinoua committed:
Merge branch 'flutter:master' into master

bambinoua committed:
Merge branch 'flutter:master' into master

bambinoua commented:
@gspencergoog could you please to help me to understand what wrong and why some checks were not successful? Initially I thought the file `tooltip.dart` was outdated and that is why I updated my repository from flutter master branch but nothing changed.

gspencergoog commented on packages/flutter/lib/src/material/popup_menu.dart:
```diff
@@ -1164,16 +1164,23 @@ class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
-          enableFeedback: enableFeedback,
-          child: widget.child,
-        ),
+    if (widget.child != null) {
```
I'm not sure why you need this code in `popup_menu.dart` now: if the tooltip is empty, it already won't show anything. No point in having similar code in two places, especially when it makes the logic more complex.

gspencergoog reviewed (commented):
That is an odd analyzer error, and one that I don't think is caused by your code changes. All I can suggest is to move to a newer version of master, since it's probably due to being synced to a bad version of master.

bambinoua committed:
Delete popup_menu.dart

As the `tooltip.dart` was changed then modification of `popup_menu.dart` is not required.

bambinoua commented:
It seems I did something wrong so I will close this pool request and open new one.
